packages/blade/src/components/ListView/_decisions/decisions.md

```diff
@@ -0,0 +1,481 @@
+- Will have bundle of Dropdown, Menu, DatePicker, BottomSheet, etc even if that particular filter is not used
+- Less intuitive because it'll introduce new config. Compared to previous API where consumers will use filters using the components that they already know how to use like Dropdown, Menu, etc.
+
+#### Extension of Table component
+
+In this case, we add quickfilters, filters, search as an option to our existing Table component
+
+Similar to [react-data-grid](https://mui.com/x/react-data-grid/filtering/quick-filter/) by MUI
+
+```jsx
+<Table
+  data={{ nodes }}
+  hasSearch
+  quickFilters={[
+    { title: 'All', onClick: (nodes) => nodes },
+    {
+      title: 'Pending',
+      onClick: (nodes) => nodes.filter((node) => node.status === 'Pending'),
+    },
+  ]}
+  filters={[
+    {
+      label: 'Status',
+      options: ['All', 'Pending', 'Completed'],
+      trigger: 'select-input',
+      onChange: ({ name, value }) => {
+        // filter on select
+      },
+    },
+    {
+      label: 'Customer Email',
+      trigger: 'text-input',
+      type: 'email',
+      onBlur: () => {
+        // filter on email change
+      },
+    },
+    {
+      label: 'Date',
+      trigger: 'date-range',
+      onChange: ({ name, value }) => {
+        // filter on date range
+      },
+    },
+  ]}
+/>
+```
+
+**Pros**
+
+- Less verbose API
+- Consistent filtering experience on all tables
+
+**Cons**
+
+- Too strict with very little flexibility
+- Adds up more things in tables which can make table difficult to maintain and will contain bundle of filters even when they are not used
+
+</details>
+
+## Enhancements / Components
+
+- Enhancements:
+  - Searchable Dropdown
+  - InputGroup (TBD)
+- New Components
+  - ListView (Layout Component)
+  - FilterChip
+
+### Dropdown with AutoComplete in Overlay
+
+<img src="image-1.png" width="300px" />
+
+```jsx
+<Dropdown>
+  <SelectInput />
+  <DropdownOverlay>
+    <DropdownHeader>
+      <AutoComplete />
+    </DropdownHeader>
+    <ActionList>
+      <ActionListItem />
+      <ActionListItem />
+      <ActionListItem />
+    </ActionList>
+</Dropdown>
+```
+
+### FilterChipGroup
+
+<img src="image-3.png" width="600px" />
+
+#### Usage
+
+```jsx
+const [duration, setDuration] = React.useState();
+const [utrNumber, setUtrNumber] = React.useState();
+const utrNumberInputValueRef = React.useRef(null);
+
+<FilterChipGroup>
+  <Dropdown>
+    <FilterChip value={duration} onClearButtonClick={({ value }) => setDuration(undefined)}>
+      Duration
+    </FilterChip>
+    <DropdownOverlay>
+      <ActionList>
+        <ActionListItem
+          title="2 days"
+          value="2d"
+          onClick={({ name, value }) => setDuration(name)}
+        />
+        <ActionListItem
+          title="1 week"
+          value="1w"
+          onClick={({ name, value }) => setDuration(name)}
+        />
+        <ActionListItem
+          title="1 month"
```

Comment:
Wouldn’t this get quite cumbersome to write all these JSX even when we just have 2-3 filter chips, not to mention most production apps will have 8-12 chips

Are there any alternatives we can approach? 🤔 Since we already know the specific inputs involved, could we consider a more streamlined, config-driven, and focused API instead?

<FilterChipInput type="select-dropdown" options={[]} />